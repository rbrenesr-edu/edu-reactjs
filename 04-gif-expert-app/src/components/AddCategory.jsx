import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewValue } ) => {


  const [inputValue, setInputValue] = useState('');
  const onInputChange = (event) => {
    setInputValue(event.target.value);
  }




  const onSubmit=(event)=>{    
    event.preventDefault();
    
    if( inputValue.trim().length <= 1 ) return;
        
    onNewValue(inputValue.trim());
    setInputValue ('');        
  }


  

  return (
    <>
        <form onSubmit={ (event) => { onSubmit(event) } } aria-label="form">
            <input
                type="test"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={ (event)=>{onInputChange(event)}  }
                />
        </form>
    </>
  )
}


AddCategory.propTypes={
  onNewValue: PropTypes.func.isRequired,
}