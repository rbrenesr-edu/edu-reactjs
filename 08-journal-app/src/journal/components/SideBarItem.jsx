import { useMemo } from "react";
import { useDispatch } from "react-redux";
import { TurnedInNot } from "@mui/icons-material";
import {
  Grid,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
} from "@mui/material";

import { setActiveNote } from "../../store/journal";

export const SideBarItem = ({ title, body, id, date, imageUrls = [] }) => {

  const dispatch = useDispatch();

  const newTitle = useMemo(() => {
    return title.length > 17 ? title.substring(0, 17) + "..." : title;
  }, [title]);

  const newBody = useMemo(() => {
    return body.length > 17 ? body.substring(0, 17) + "..." : body;
  }, [body]);


  const onClickNote = () =>{    
    dispatch(setActiveNote({id, title, body, date, imageUrls}))
  }

  return (
    <ListItem disablePadding onClick={onClickNote}>
      <ListItemButton>
        <ListItemIcon>
          <TurnedInNot />
        </ListItemIcon>
        <Grid container>
          <ListItemText primary={newTitle} />
          <ListItemText secondary={newBody} />
        </Grid>
      </ListItemButton>
    </ListItem>
  );
};
