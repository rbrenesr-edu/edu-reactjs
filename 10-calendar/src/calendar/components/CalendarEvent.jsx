
export const CalendarEvent = ({ event }) => {

    const { title, notes, user } = event;

  return (
    <>
        <strong>{ title }</strong>
        <p>- { notes }</p>
    </>
  )
}
