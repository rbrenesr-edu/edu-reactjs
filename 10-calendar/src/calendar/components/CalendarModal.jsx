import { useState } from 'react';
import Modal from 'react-modal'

const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
    },
  };

Modal.setAppElement('#root');

export const CalendarModal = () => {

const [isModalOpen, setIsModalOpen] = useState(true)


const  onCloseModal = () => { 
    console.log("cerrando modal"); 
    setIsModalOpen(false);
}


  return (
    <Modal
        isOpen={ isModalOpen }        
        onRequestClose={ onCloseModal }
        style={customStyles}
        contentLabel="Example Modal"
        className="modal"
        overlayClassName="modal-fondo"
        closeTimeoutMS={ 200 }
    >

        <h1>Modal</h1>
        <hr />
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Earum, ex!
        </p>

    </Modal>
  )
}
