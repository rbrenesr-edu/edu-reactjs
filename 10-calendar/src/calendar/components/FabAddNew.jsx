import { addHours } from "date-fns";
import { useUiStore, useCalendarStore } from "../../hooks"
import {  } from "../../hooks/useUiStore"

export const FabAddNew = () => {

const { openDateModal } = useUiStore();
 const { setActiveEvents } =  useCalendarStore();

const handleClickNew=()=>{
  openDateModal();
  setActiveEvents(
    {      
      title: "",
      notes: "",
      start: new Date(),
      end: addHours(new Date(), 2),
      bgColor: "#fafafa",
      user: {
        id: "123465",
        name: "Rafael Brenes",
      }
    }
  );
}

  return (
    <button
        className="btn btn-primary fab"
        onClick={ handleClickNew }
    >
        <i className="fas fa-plus" ></i>
    </button>
  )
}
