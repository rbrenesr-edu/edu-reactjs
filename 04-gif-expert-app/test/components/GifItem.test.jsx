import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test GifItem', () => { 

    const titulo = 'Titulo';
    const url = 'http://localhost/ulr_dir';

    test('Match con el snapshot', () => { 

        const { container } = render( <GifItem title={ titulo } url={ url }/> );
        expect( container ).toMatchSnapshot();
     })

     test('Debe mostrar la imagen y el url indicado', () => { 
        //console.log({ titulo, url } );
        render( <GifItem title={ titulo } url={ url }/> );
        //screen.debug();  
        //expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

      });

      test('Debe de mostrar el tÃ­tulo en el componente', () => { 
        screen.debug();  
        render( <GifItem title={ titulo } url={ url }/> );
        expect( screen.getByText( titulo ) ).toBeTruthy();

       })

 })