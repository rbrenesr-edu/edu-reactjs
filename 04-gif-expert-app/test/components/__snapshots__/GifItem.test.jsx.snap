// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem Match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="http://localhost/ulr_dir"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
